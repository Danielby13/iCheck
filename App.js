import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { UserProvider } from "./src/context/UserContext";
import { FirebaseProvider } from "./src/context/FirebaseContext";
import AppStackScreens from "./src/stacks/AppStackScreens";
import FlashMessage from "react-native-flash-message";
export default App = () => {
  return (
    <FirebaseProvider>
      <UserProvider>
        <NavigationContainer>
          <AppStackScreens />
        </NavigationContainer>
        <FlashMessage position="top" floating={true} />
      </UserProvider>
    </FirebaseProvider>
  );
};
