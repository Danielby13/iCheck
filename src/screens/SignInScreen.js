// import { StatusBar } from 'expo-status-bar';
import React, { useState, useContext } from "react";
import styled from "styled-components";
import Text from "../components/Text";
import { FirebaseContext } from "../context/FirebaseContext";
import { UserContext } from "../context/UserContext";
import { Image, Linking, Button } from "react-native";

export default SignInScreen = ({ navigation }) => {
  // sign in screen
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const [loading, setLoading] = useState(false);
  const firebase = useContext(FirebaseContext);
  const [_, setUser] = useContext(UserContext);

  const signIn = async () => {
    // get user info to log in from the db
    setLoading(true);

    try {
      await firebase.signIn(email, password);

      const uid = firebase.getCurrentUser().uid;
      const userInfo = await firebase.getUserInfo(uid);

      setUser({
        firstName: userInfo.firstName,
        email: userInfo.email,
        uid,
        isLoggedIn: true,
      });
    } catch (error) {
      alert(error.message);
    } finally {
      setLoading(false);
    }
  };
  return (
    <Container>
      <Main>
        <Text title semi center>
          ברוכים הבאים
        </Text>
      </Main>

      <Auth>
        <AuthContainer>
          <AuthTitle>כתובת אימייל</AuthTitle>
          <AuthField
            autoCapotalize="none"
            autoCompleteType="email"
            autoCorrect={false}
            autoFocus={true}
            keyboardType="email-address"
            onChangeText={(email) => setEmail(email.trim())}
            value={email}
          />
        </AuthContainer>

        <AuthContainer>
          <AuthTitle>סיסמה</AuthTitle>
          <AuthField
            autoCapotalize="none"
            autoCompleteType="password"
            autoCorrect={false}
            secureTextEntry={true}
            onChangeText={(password) => setPassword(password.trim())}
            value={password}
          />
        </AuthContainer>
      </Auth>

      <SignInContainer onPress={signIn} disabled={loading}>
        {loading ? (
          <Loading />
        ) : (
          <Text bold center color="#ffffff">
            כניסה
          </Text>
        )}
      </SignInContainer>

      <SignUp onPress={() => navigation.navigate("SignUp")}>
        <Text small center>
          לא רשום?{" "}
          <Text medium bold color="#54c9cd">
            {" "}
            להרשמה{" "}
          </Text>
        </Text>
      </SignUp>

      <HeaderGraphic>
        <RightCircle />
        {/* <LeftCircle /> */}
        <HeaderLogo>
          <Image
            source={require("../../assets/logo.png")}
            style={{ width: 300, height: 89 }}
          />
        </HeaderLogo>
      </HeaderGraphic>

      <StatusBar barStyle="light-content" />
    </Container>
  );
};

const Container = styled.View`
  flex: 1;
`;
const Main = styled.View`
  margin-top: 192px;
`;
const Auth = styled.View`
  margin: 64px 32px 32px;
`;

const AuthContainer = styled.View`
  margin-bottom: 32px;
`;

const AuthTitle = styled(Text)`
  color: #8e93a1;
  font-size: 13px;
  text-transform: uppercase;
  font-weight: 800;
  text-align: left;
`;

const AuthField = styled.TextInput`
  border-bottom-color: #8e93a1;
  border-bottom-width: 0.5px;
  height: 48px;
`;

const SignInContainer = styled.TouchableOpacity`
  margin: 0 32px;
  height: 48px;
  align-items: center;
  justify-content: center;
  background-color: #54c9cd;
  border-radius: 6px;
`;

const Loading = styled.ActivityIndicator.attrs((props) => ({
  color: "#ffffff",
  size: "small",
}))``;

const SignUp = styled.TouchableOpacity`
  margin-top: 16px;
`;

const HeaderGraphic = styled.View`
  position: absolute;
  width: 100%;
  top: -50px;
  z-index: -100;
`;

const RightCircle = styled.View`
  background-color: #ddb0ce;
  position: absolute;
  width: 400px;
  height: 450px;
  border-radius: 200px;
  left: -100px;
  top: -200px;
`;
// background-color: #4abec2;
const LeftCircle = styled.View`
  background-color: #acacac;
  position: absolute;
  width: 200px;
  height: 200px;
  border-radius: 100px;
  left: -50px;
  top: -50px;
`;

const HeaderLogo = styled.View`
  left: 45px;
  top: 130px;
`;

const StatusBar = styled.StatusBar``;

// const styles = StyleSheet.create({
//   container: {
//     flex: 1,
//     alignItems: 'center',
//     justifyContent: 'center',
//   },
// });
