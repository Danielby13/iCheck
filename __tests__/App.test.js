/**
 * @jest-environment jsdom
 */
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import SignInScreen from "../src/screens/SignInScreen";
import { render, fireEvent } from "@testing-library/react-native";
jest.useFakeTimers();
jest.mock("react-native/Libraries/Animated/src/NativeAnimatedHelper");
jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});
jest.mock("react-native-elements");
jest.mock("firebase/firestore");
configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly", () => {
  renderer.create(<App />);
});

it("should render the Sign In Screen without crashing", () => {
  const rendered = renderer.create(<SignInScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Sign Up Screen without crashing", () => {
  const rendered = renderer.create(<SignUpScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Share All Screen without crashing", () => {
  const rendered = renderer.create(<ShareAllScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Button Text Element", () => {
  const wrapper = shallow(<ShareAllScreen />);
  expect(wrapper.find("Button")).toHaveLength(3);
});

it("should render the Invoice History Screen without crashing", () => {
  const rendered = renderer.create(<ResetAllScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Change Personal Details Screen without crashing", () => {
  const rendered = renderer.create(<ChangePersonalDetailsScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Delete User Screen without crashing", () => {
  const rendered = renderer.create(<DeleteUserScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Terms Screen without crashing", () => {
  const rendered = renderer.create(<terms />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Terms Screen without crashing", () => {
  const rendered = renderer.create(<SecurityPolicyScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the FAQ Screen without crashing", () => {
  const rendered = renderer.create(<FAQScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render the Update Password Screen without crashing", () => {
  const rendered = renderer.create(<updatePasswordScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render Update Personal Info Screen without crashing", () => {
  const rendered = renderer.create(<updatePersonalInfo />).toJSON();
  expect(rendered).toBeTruthy();
});

it("should render Invoice History Screen without crashing", () => {
  const rendered = renderer.create(<InvoiceHistory />).toJSON();
  expect(rendered).toBeTruthy();
});
