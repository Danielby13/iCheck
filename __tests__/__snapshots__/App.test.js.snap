// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              needsOffscreenAlphaCompositing={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#222222",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <Text
                          color="#FFFFFF"
                          style={
                            Array [
                              Object {
                                "color": "#FFFFFF",
                                "fontSize": 32,
                                "fontWeight": "400",
                                "marginBottom": 0,
                                "marginLeft": 0,
                                "marginRight": 0,
                                "marginTop": 0,
                                "paddingBottom": 0,
                                "paddingLeft": 0,
                                "paddingRight": 0,
                                "paddingTop": 0,
                                "textAlign": "right",
                              },
                            ]
                          }
                          title={true}
                        >
                          iCheck App
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaProvider>
  </View>,
  <View
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": -48,
          },
        ],
        "zIndex": 99,
      }
    }
  />,
]
`;
